webcompat/helpers.py

```diff
@@ -407,6 +408,53 @@ def wrapped_func(*args, **kwargs):
     return wrapped_func
 
 
+def mockable_report(func):
+    """Mock out an API reponse for `issues.report_issue`, if in TESTING mode.
+
+    This allows us to test form submission, always returning a special
+    issue 0 (that doesn't exist). It loads fixture data from a "zero.json"
+    template, and merges in the reported title and body, saving it to "0.json".
+
+    Note: unless you've created 0.json locally by running tests, navigating
+    there directly will 404. We don't check it into source control because
+    it can change depending on the test. But only tests should care abou this.
+    """
+    class Mocked201Response:
```

Comment:
This creates a mocked response for functional tests when app.config['TESTING'].
I wonder what will be the outcome for unit-testing. I'm slightly uncomfortable. I have the feeling we are doing something wrong when our main code (outside of tests/) becomes peppered little by little by tests related things.

https://github.com/webcompat/webcompat.com/blob/fa99d5d35699911ed2cecad59e476f9f295feade/webcompat/helpers.py#L381-L407

https://github.com/webcompat/webcompat.com/blob/7517d8c1f25b9f70eca1536e240bd2f6ce69b95b/webcompat/views.py#L83-L87

https://github.com/webcompat/webcompat.com/blob/7517d8c1f25b9f70eca1536e240bd2f6ce69b95b/webcompat/views.py#L110-L111

The goal if I understand is to cut out the request to GitHub?

I don't have a solution right out of my hat :) 
but https://github.com/domenic/sinon-chai with http://sinonjs.org/
seems to be something to look at.

Reply:
http://sinonjs.org/releases/v4.5.0/fake-xhr-and-server/

Reply:
> I have the feeling we are doing something wrong when our main code (outside of tests/) becomes peppered little by little by tests related things.

I think I agree with you that it's a code smell.

Reply:
Hm, so we could either land this now, and file a follow up bug to remove testing special-cased codepaths from everywhere. Let me investigate a bit first.